examples/secret_storage/secret_storage.go

```diff
@@ -0,0 +1,572 @@
+/*
+WHAT IS IN THIS EXAMPLE?
+
+Keybase has added an encrypted key-value store intended to support
+security-conscious bot development with persistent state. It is a
+place to store small bits of data that are
+  (1) encrypted for a team or user (via the user's implicit self-team: e.g.
+alice,alice),
+  (2) persistent across logins
+  (3) fast and durable.
+
+It supports putting, getting, listing, and deleting. A team has many
+namespaces, a namespace has many entryKeys, and an entryKey has one current
+entryValue. Namespaces and entryKeys are in cleartext, and the Keybase client
+service will encrypt and sign the entryValue on the way in (as well as
+decrypt and verify on the way out) so keybase servers cannot see it or
+forge it.
+
+-----------
+
+This example implements a simple bot to manage hackerspace tool rentals. It
+shows one way you can obfuscate entryKeys (which are not encrypted) by
+storing their HMACs, so that no one but your team (not even
+Keybase) can know about the names of all the cool tools you have; you can do
+something similar to hide namespaces.
+
+Additionally this example handles concurrent writes by using explicit revision
+numbers to prevent one user from unintentionally clobbering another user's
+rental updates.
+
+*/
+package main
+
+import (
+	"crypto/hmac"
+	"crypto/rand"
+	"crypto/sha256"
+	"encoding/hex"
+	"encoding/json"
+	"flag"
+	"fmt"
+	"log"
+	"os"
+	"strings"
+
+	"github.com/keybase/go-keybase-chat-bot/kbchat"
+	"github.com/keybase/go-keybase-chat-bot/kbchat/types/keybase1"
+
+	"golang.org/x/sync/errgroup"
+)
+
+// SecretKeyKVStoreAPI is a KVStoreAPI that hides the entryKeys from Keybase servers.
+// It does so by HMACing entryKeys using a per-(team, namespace) secret,
+// and storing the HMAC instead of the plaintext entryKey. This approach
+// does not handle any secret rotation, and does not expect the secret to
+// change.
+//
+// The plaintext entryKey is stored in it's corresponding JSON entryValue
+// under the key "_key" to enable listing.
+//
+// This approach does not hide memory access patterns. Also, Keybase
+// servers prevent a removed team member from continuing to access a team's
+// data, but if that were somehow bypassed*, a former team member who still
+// knows the HMAC secret can check for the presence of specific entryKeys
+// (*but you probably have bigger issues to deal with in that case...).
+type SecretKeyKVStoreAPI struct {
+	api       kbchat.KVStoreAPI
```

Comment:
NAB: suggest calling this something other than `api`. maybe `client`?

Reply:
only using API because the rest of the code uses the word API (see `type API`)